### PR TITLE
Fix issue #15269

### PR DESCRIPTION
**Pull request by adamcavendish**
A possible solution to fix issue #15269.

### CODE REVIEW

**Comment by drpngx** [comment]
@adamcavendish could you sign the CLA?

**Comment by adamcavendish** [comment]
I signed it!

**Comment by adamcavendish** [comment]
@drpngx Thx for reminding me of that

**Comment by drpngx** [comment]
Jenkins, test this please.

**Comment by martinwicke** [comment]
I'm afraid of this, especially given how unpredictable the variable_scope reuse behavior is. @fchollet  @lukaszkaiser do you have a good enough understanding of this to be confident this does not break things in some obscure way? See more discussion on the issue #15269.

**Comment by fchollet** [comment]
Thanks for the PR!

@martinwicke I don't know, but let's find out.

@adamcavendish without new unit tests, it is impossible to assess the effect of the fix or whether it does fix the issue. Please add a series of unit tests that assert that the new behavior is the one we expect (see discussion). These should be tests that would have been previously failing and would now be passing.

**Comment by lukaszkaiser** [comment]
I second @fchollet : let's first add tests to make clear what's happening.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?

**Comment by drpngx** [comment]
@adamcavendish any chances to write the tests?

**Comment by adamcavendish** [comment]
@drpngx Sure, but it does change things, especially `variable_scope` is a stable API. Like what I've said in the issue #15269 , it'd change `batch_norm` layer's incremental naming. Therefore, what kind of UT would you expect? Simply adding my issue's example or some further?

**Comment by drpngx** [comment]
@fchollet and @lukaszkaiser can chime in. You need to show exercise expected behavior.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by fchollet** [comment]
My comment is still valid and has not been acted on:

"without new unit tests, it is impossible to assess the effect of the fix or whether it does fix the issue. Please add a series of unit tests that assert that the new behavior is the one we expect (see discussion). These should be tests that would have been previously failing and would now be passing."

Since it has been a while, I consider the PR abandoned and I will close the PR. Feel free to reopen later, with unit tests.
